[Documentation] DQT module Readme (#5206)
aces/Loris: modules/dataquery/readme.md

## Other Notes
The most closely related module to the DQT is the Data Release module, which is used to show snapshots of data for any given release. 

## Permissions
Users must have `dataquery_view` permission to use the DQT. It’s important to note that more granular permissions do not exist for this module, to restrict access to a subset of the data. Anyone with the `dataquery_view` permission has access to the DQT, and therefore has access to all data in the LORIS instance. 

## Interactions with LORIS
The CouchDB import scripts (those named CouchDB_Import_*.php in the LORIS tools directory) must be run to import data from MySQL to CouchDB before the DQT can be used. It is recommended to set up a cronjob to execute these scripts to refresh the data on a regular basis. 

## Configuration
Use of the DQT module requires setting up the CouchDB server and credentials:

[CouchDB](http://couchdb.apache.org)

> Note: Your version of Erlang should be the most recent version that is compatible with both your CouchDB.

Reviewer: `that is compatible with both your CouchDB` isn't grammatically correct.. I think we can probably just remove this note.

Reviewer 2: 👍  vote to remove this line

Author: Removed

Once CouchDB is installed, follow these steps to complete setup:

1. create a database on your local CouchDB instance
2. clone the code from the server [http://couchdb.loris.ca:5984/dataquerytool-$VERSION]
..* ensure $VERSION is separated by underscores rather than dots (because dots are not allowed in CouchDB database names)
..* i.e.
`curl -H 'Content-Type: application/json' -X POST http://$YOURCOUCHDBADMIN:$YOURCOUCHADMINPASS@$YOURSERVERNAME:5984/_replicate -d '{"source":"http://couchdb.loris.ca:5984/dataquerytool-1_0_0", "target":"$YOURDATABASENAME"}'`

Finally, there are further steps required to complete configuration:

Amend the section of your LORIS config.xml
`<CouchDB>
    <SyncAccounts>true</SyncAccounts>
    <database>dqg</database>
    <hostname>localhost</hostname>
    <port>5984</port>
    <admin>adminuser</admin>
    <adminpass>adminpass</adminpass>
</CouchDB>`

> To load the Data Query Tool with data stored in LORIS, run the `CouchDB_Import_*` scripts, in tools/ directory: 

Reviewer 3: ```suggestion
To load the Data Query Tool with data stored in LORIS, run the `CouchDB_Import_*` scripts, in tools/ directory: 

```

## Other Notes
The most closely related module to the DQT is the Data Release module, which is used to store snapshots of data for any given release. 

## Permissions
Users must have `dataquery_view` permission to use the DQT. 

It’s important to note that more granular permissions do not exist for this module, to restrict access to a subset of the data. Anyone with the `dataquery_view` permission has access to view, query and download data from the DQT, and therefore has access to all data loaded in the DQT from this LORIS instance.

## Interactions with LORIS
Before the DQT can be used, the CouchDB import scripts must be run to load data from LORIS' MySQL tables into CouchDB. These scripts are found in the `tools/` directory, named `CouchDB_Import_*.php`.

It is recommended to set up a cronjob to execute these scripts to refresh the data on a regular basis. 

## Configuration
Use of the DQT module requires setting up the CouchDB server and credentials:

> [CouchDB](http://couchdb.apache.org)

Reviewer: This should also link to https://github.com/aces/Data-Query-Tool as a requirement, I think.

Author: Isn't that link just a repeat of the info laid out in this readme?

Author: Isn't everything in that link addressed in this readme?

Reviewer 2: It may be less than 100% up-to-date but still contains tips for developers (vs. those just installing the end product)

Author: gotcha, thanks! I added a "note" at the end of config.

Reviewer 2: 👍

## Intended Users
The DQT module is intended for researchers who wish to download data stored in LORIS. Users can access only internal data that belongs to their LORIS instance. 

## Scope
The module implements a proxy to pass through queries to the Loris CouchDB DQT backend, which must be setup independently (see Configuration section).

It does NOT alter data in any way other than that required to convert the data from the MySQL to the CouchDB format. In particular, it does no anonymization or filtering of participant data that is already in LORIS. This must be done independently, before distributing data externally.

## Other Notes
The most closely related module to the DQT is the Data Release module, which is used to store snapshots of data for any given release. 

## Permissions
Users must have `dataquery_view` permission to use the DQT. 

> It’s important to note that more granular permissions do not exist for this module, to restrict access to a subset of the data. Anyone with the `dataquery_view` permission has access to view, query and download data from the DQT, and therefore has access to all data loaded in the DQT from this LORIS instance.

Reviewer 3: ```suggestion
It’s important to note that granular permissions do not exist for this module. Anyone with the `dataquery_view` permission has access to view, query and download data from the DQT, and therefore has access to all data loaded in the DQT from this LORIS instance. It is not possible to restrict user access to a subset of the data.
```

Small grammar improvement.

## Interactions with LORIS

> Before the DQT can be used, the CouchDB import scripts must be run to load data from LORIS' MySQL tables into CouchDB. These scripts are found in the `tools/` directory, named `CouchDB_Import_*.php`.

Reviewer 3: ```suggestion
Before the DQT can be used, the CouchDB import scripts must be run in order to load data from LORIS's MySQL tables into CouchDB. These scripts are found in the `tools/` directory, named `CouchDB_Import_*.php`.
```

Reviewer 2: Agree - @knoll-alex could you commit this suggested tweak?

2. Clone the code from our server:

`
curl -H 'Content-Type: application/json' -X POST http://$YOURCOUCHDBADMIN:$YOURCOUCHADMINPASS@$YOURSERVERNAME:5984/_replicate -d '{"source":"http://couchdb.loris.ca:5984/dataquerytool-1_0_0", "target":"$YOURDATABASENAME"}'
`

3. Amend the `<CouchDB>` section of your LORIS `project/config.xml` as follows:

```
<CouchDB>
    <database>dqg</database>
    <hostname>localhost</hostname>
    <port>5984</port>
    <admin>adminuser</admin>
    <adminpass>adminpass</adminpass>
</CouchDB>
```

4. To load the Data Query Tool with data stored in LORIS, run the `CouchDB_Import_*` scripts, in the `tools/` directory: 

`cd $lorisroot/tools`

##### Import the base candidate data

`php CouchDB_Import_Demographics.php`

##### Import the LORIS instrument data
This step is optional and not required if only the MRI portion of LORIS is used:

`php CouchDB_Import_Instruments.php`

##### Import the LORIS MRI data
This step is optional and not required if the MRI portion of LORIS isn't installed:

`php CouchDB_Import_MRI.php`

#### NOTE - for developer setup tips, see further instructions [here](https://github.com/aces/Data-Query-Tool)

>     <port>5984</port>

Reviewer 3: I'm not quite sure why this line is here.

## Other Notes
The most closely related module to the DQT is the Data Release module, which is used to show snapshots of data for any given release. 

## Permissions
Users must have `dataquery_view` permission to use the DQT. It’s important to note that more granular permissions do not exist for this module, to restrict access to a subset of the data. Anyone with the `dataquery_view` permission has access to the DQT, and therefore has access to all data in the LORIS instance. 

## Interactions with LORIS
The CouchDB import scripts (those named CouchDB_Import_*.php in the LORIS tools directory) must be run to import data from MySQL to CouchDB before the DQT can be used. It is recommended to set up a cronjob to execute these scripts to refresh the data on a regular basis. 

## Configuration
Use of the DQT module requires setting up the CouchDB server and credentials:

[CouchDB](http://couchdb.apache.org)

> [Erica](https://github.com/benoitc/erica)

Reviewer: Erica isn't required if the hard way is removed

Author: deleted!

Note: Your version of Erlang should be the most recent version that is compatible with both your CouchDB and Erica.

Once these are installed, there are two ways to complete setup. See below:

#### The Easy Way (for non-developers)

1. create a database on your local CouchDB instance
2. clone the code from the server [http://couchdb.loris.ca:5984/dataquerytool-$VERSION]
..* ensure $VERSION is separated by underscores rather than dots (because dots are not allowed in CouchDB database names)
..* i.e.
`curl -H 'Content-Type: application/json' -X POST http://$YOURCOUCHDBADMIN:$YOURCOUCHADMINPASS@$YOURSERVERNAME:5984/_replicate -d '{"source":"http://couchdb.loris.ca:5984/dataquerytool-1_0_0", "target":"$YOURDATABASENAME"}'`

> #### The Hard Way (for developers)

Reviewer: I don't think "The Hard Way" should be here. It's in the README of the aces/Data-Query-Tool repo, and only required for developers. The code that you need to know how to do that for also hasn't been touched in years.

Author: deleted!

## Intended Users
The DQT module is intended for researchers who wish to download data stored in LORIS. Users can access only internal data that belongs to their LORIS instance. 

## Scope
The module implements a proxy to pass through queries to the Loris CouchDB DQT backend, which must be setup independently (see Configuration section).

It does NOT alter data in any way other than that required to convert the data from the MySQL to the CouchDB format. In particular, it does no anonymization or filtering of participant data that is already in LORIS. This must be done independently, before distributing data externally.

## Other Notes
The most closely related module to the DQT is the Data Release module, which is used to store snapshots of data for any given release. 

## Permissions

> Users must have `dataquery_view` permission to use the DQT. It’s important to note that more granular permissions do not exist for this module, to restrict access to a subset of the data. Anyone with the `dataquery_view` permission has access to the DQT, and therefore has access to all data in the LORIS instance. 

Reviewer 2: Let's leave these unresolved so that the original change-requester (me in this case) can verify - (i.e. leave them a little time to resolve the conversation)

## Configuration
Use of the DQT module requires setting up the CouchDB server and credentials:

[CouchDB](http://couchdb.apache.org)

Note: Your version of Erlang should be the most recent version that is compatible with both your CouchDB.

Once CouchDB is installed, follow these steps to complete setup:

1. Create a database on your local CouchDB instance
2. Clone the code from the server [http://couchdb.loris.ca:5984/dataquerytool-$VERSION]
..* ensure $VERSION is separated by underscores rather than dots (because dots are not allowed in CouchDB database names)
..* i.e.

`
curl -H 'Content-Type: application/json' -X POST http://$YOURCOUCHDBADMIN:$YOURCOUCHADMINPASS@$YOURSERVERNAME:5984/_replicate -d '{"source":"http://couchdb.loris.ca:5984/dataquerytool-1_0_0", "target":"$YOURDATABASENAME"}'
`

Finally, there are further steps required to complete configuration:

Amend the `<CouchDB>` section of your LORIS `project/config.xml` as follows:

```
<CouchDB>
    <SyncAccounts>true</SyncAccounts>
    <database>dqg</database>
    <hostname>localhost</hostname>
    <port>5984</port>
    <admin>adminuser</admin>
    <adminpass>adminpass</adminpass>
</CouchDB>

> ```

Reviewer: That doesn't exist anymore, the `<SyncAccounts>` part should be removed.

> ```

Author: Removed!

> ```

Author: Removed

> ```

Reviewer 2: resolving 👍

# Data Query Tool

## Purpose

> The Data Query Tool (“The DQT”) is used to query and export data stored in LORIS. Designed as a front-end tool, it is located within the Reports drop-down menu. 

Reviewer 3: ```suggestion
The Data Query Tool (“The DQT”) is used to query and export data stored in LORIS. It is located within the Reports drop-down menu. 
```

Reviewer 2: makes sense. 👍

## Interactions with LORIS

Before the DQT can be used, the CouchDB import scripts must be run in order to load data from LORIS's MySQL tables into CouchDB. These scripts are found in the `tools/` directory, named `CouchDB_Import_*.php`.

It is recommended to set up a cronjob to execute these scripts to refresh the data on a regular basis. 

## Configuration

Use of the DQT module requires setting up the CouchDB server and credentials:

[CouchDB](http://couchdb.apache.org)

Once CouchDB is installed, follow these steps to complete setup:

1. Create a database on your local CouchDB instance

2. Clone the code from our server:

`
curl -H 'Content-Type: application/json' -X POST http://$YOURCOUCHDBADMIN:$YOURCOUCHADMINPASS@$YOURSERVERNAME:5984/_replicate -d '{"source":"http://couchdb.loris.ca:5984/dataquerytool-1_0_0", "target":"$YOURDATABASENAME"}'
`

3. Amend the `<CouchDB>` section of your LORIS `project/config.xml` as follows:

> ```

Reviewer 3: ```suggestion
```xml
```

This should make the formatting a little nicer when rendered in GitHub.
